src/box/lua/xlog.c

```diff
@@ -365,8 +365,63 @@ lbox_xlog_parser_open_pairs(struct lua_State *L)
 	return 3;
 }
 
+/** Opens an xlog file and returns its meta block. */
+static int
+lbox_xlog_parser_open_meta(struct lua_State *L)
+{
+	int args_n = lua_gettop(L);
+	if (args_n != 1 || !lua_isstring(L, 1))
+		luaL_error(L, "Usage: parser.open(log_filename)");
+
+	const char *filename = luaL_checkstring(L, 1);
+
+	/* Construct xlog cursor and object */
+	struct xlog_cursor *cur = xcalloc(1, sizeof(struct xlog_cursor));
+	if (xlog_cursor_open(cur, filename) < 0) {
+		return luaT_error(L);
+	}
+	if (strncmp(cur->meta.filetype, "SNAP", 4) != 0 &&
```

Comment:
IMO there's no need to hardcode filetype checking here. If `xlog_cursor_open` managed to open the xlog, then simply pass all the parsed meta to the user.

Reply:
Removed the filetype validation. I hesitated a lot, whether I should include them or not.